tests/FlexTest.php

```diff
@@ -78,6 +97,28 @@ public function getRecordTests()
         ];
     }
 
+    /**
+     * dataset0: custom COMPOSER env is set, generates correct symfony lock
+     * dataset1: nothing overriden, just verify default names
+     * dataset2: composer = default, custom symfony.lock path
+     * @testWith ["composer.beta.json","", "composer.beta.lock", "symfony.beta.lock"]
```

Comment:
Why not use a classic data provider?

Reply:
i prefer annotations over dataprovider, i think its lesser, code and easier to maintain/read
you require me to change it? - is it a ko-criteria?

Reply:
I don't think it is less code (because your annotation contains code here). And I find maintenance easier when the code is written in actual code rather than instead a comment, as tooling will be able to recognize it and help me maintain it (I will get code highlighting, autocompletion, etc...)

and regarding readability, I find data providers quite readable when using a generator to provide the different datasets (the syntax using a nested array is indeed harder to read)

Reply:
ok got you, your repo, your rules, and all valid points, changed to a dataprovider.